frame/privilege/src/lib.rs

```diff
@@ -148,7 +161,70 @@ pub mod pallet {
 	// FIXME: Temporary fix to get CI to pass, separate PRs will be made per pallet to refactor to
 	// use OptionQuery instead
 	#[allow(clippy::disallowed_types)]
-	pub type GroupCount<T: Config> = StorageValue<_, u32, ValueQuery>;
+	pub type GroupCount<T: Config> = StorageValue<
+		_,
+		u32,
+		ValueQuery,
+		Counter<ZeroInit, IncrementToMax<T::MaxGroup, TooManyGroup, Error<T>>, SafeDecrement>,
+	>;
+
+	error_to_pallet_error!(
```

Comment:
overall ok, so I dislike macro. copy paste TooManyGroup 3 times around (macro 2, Error<T> 1), it kind of flat outs error flat into mod. 

better map_err on each counter in case of overflow.


```
error_mod!(
#[pallet::error]
	pub enum Error<T> {
           Foo,
           Bar,
         }

mod error {
   struct Foo(());
   struct Bar(());
}

```

Reply:
The idea was to remove the need for `map_err`, since it's just boilerplate. I can try to think of a different approach, but I really think this is the best bet.

Perhaps having the macro just take in one ident?

```rust
error_to_pallet_error!(
    TooManyGroup,
)
```

It would force the name of the IncrementToMax error but I think that's ok, since it's all internal anyways.

Reply:
ok oke if `error_mod` will generate for each Error type, and type with private constructor (i.e. you cannot create instance of type).

Reply:
so I am not sure about having macro which needs to map errors manually additionally.

Reply:
How about this?

```rust
error! {
    #[pallet::error]
    #[derive(PartialEqNoBound)]
    pub enum Error<T> {
        /// hello
        #(error)
        ZeroInitIncrementToMaxValueTooLarge,
        OneInitIncrementToMaxValueTooLarge,
        DefaultInitIncrementToMaxValueTooLarge,
    }
}
```

Expanding to:
```rust
pub mod macros {
    #[pallet::error]
    pub enum Error<T> {
        /// hello
        ZeroInitIncrementToMaxValueTooLarge,
        OneInitIncrementToMaxValueTooLarge,
        DefaultInitIncrementToMaxValueTooLarge,
    }

    pub struct ZeroInitIncrementToMaxValueTooLarge;

    impl<T: Config> From<ZeroInitIncrementToMaxValueTooLarge> for Error<T> {
        fn from(_: ZeroInitIncrementToMaxValueTooLarge) -> Error<T> {
            Error::<T>::ZeroInitIncrementToMaxValueTooLarge
        }
    }
}
```

The `#(error)` is arbitrary, all that's needed is some sort of marker. (`#[error]` can't be used due to parsing ambiguities with doc comments. It could be done, but I don't really want to write a TT muncher.)